lib/Net/Async/Blockchain/Block.pm

```diff
@@ -0,0 +1,115 @@
+package Net::Async::Blockchain::Block;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+Net::Async::Blockchain::Block - Block abstraction.
+
+=head1 SYNOPSIS
+
+Objects of this type would not normally be constructed directly.
+
+=head1 DESCRIPTION
+
+Block abstraction
+
+=over 4
+
+=back
+
+=cut
+
+no indirect;
+
+sub message_type : method { shift->{message_type} }
+sub currency : method     { shift->{currency} }
+sub number : method       { shift->{number} }
```

Comment:
as long as  I know, we are using the `method` word when the function name is a reserved word, so I believe no need for method word here